standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/txn/TxnHandler.java

```diff
@@ -1339,7 +1340,8 @@ public void commitTxn(CommitTxnRequest rqst)
             // corresponding open txn event.
             LOG.info("Target txn id is missing for source txn id : " + sourceTxnId +
                     " and repl policy " + rqst.getReplPolicy());
-            return;
+            throw new NoSuchTxnException("Source transaction: " + JavaUtils.txnIdToString(sourceTxnId)
```

Comment:
Add target txn id as well.

Reply:
target txn id is not present. that's the reason this exception is being thrown.

Reply:
Aren't we aborting based on target txn id so we know which target txn id we are looking for?

Reply:
no, we have only source txn id and the replPolicy. Using this info, we query the REPL_TXN_MAP table and get corresponding target txn id and then commit this txn.

Reply:
Didn't get this. Don't we do "abort txn  <txnId>" here the txn  id is of target and not of source, right?

Reply:
abort txn txnId would not have replPolicy set. So, its execution would not come here.